tuxemon/save.py

```diff
@@ -52,127 +54,172 @@
+		else:
+			text = json.dumps(save_data, indent=4, separators=(",", ": "))
+		with open(save_path, "w") as f:
+			logger.info("Saving data to save file: " + save_path)
+			# Don't dump straight to the file: if we crash it would corrupt
+			# the save_data
+			f.write(text)
+	#text = json.dumps(save_data, indent=4, separators=(",", ": "))
 
 
 def load(slot: int) -> Optional[Mapping[str, Any]]:
-    """
-    Loads game state data from a save file.
-
-    Parameters:
-        slot: The save slot to load game data from.
-
-    Returns:
-        Dictionary containing game data to load.
-
-    """
-    save_path = f"{prepare.SAVE_PATH}{slot}.save"
-    save_data = open_save_file(save_path)
-    if save_data:
-        return upgrade_save(save_data)
-    elif save_data is None:
-        # File not found; it probably wasn't ever created, so don't panic
-        return None
-    else:
-        save_data["error"] = "Save file corrupted"
-        save_data["player_name"] = "BROKEN SAVE!"
-        logger.error("Failed loading save file.")
-        return save_data
-
-
-def open_save_file(save_path: str) -> Optional[Dict[str, Any]]:
-    try:
-        with open(save_path) as save_file:
-            try:
-                return json.load(save_file)
-            except ValueError as e:
-                logger.error("cannot decode JSON: %s", save_path)
-
-            try:
-                return cbor.load(save_file)
-            except ValueError as e:
-                logger.error("cannot decode save CBOR: %s", save_path)
-
-            return {}
-
-    except OSError as e:
-        logger.info(e)
-        return None
-
+	"""
+	Loads game state data from a save file.
+
+	Parameters:
+		slot: The save slot to load game data from.
+
+	Returns:
+		Dictionary containing game data to load.
+
+	"""
+	if config.compress_save:
+		if config.csave_lzma:
+			save_path = f"{prepare.SAVE_PATH}{slot}.csave.lza"
+		else:
+			save_path = f"{prepare.SAVE_PATH}{slot}.csave.gz"
+	else:
+		save_path = f"{prepare.SAVE_PATH}{slot}.save"
+
+	save_data = open_save_file(save_path)
+
+	# DEBUG STUFF HERE
+	#print("ayo look here")
+	#print(save_data)
+
+	if save_data:
+		return upgrade_save(save_data)
+	elif save_data is None:
+		# File not found; it probably wasn't ever created, so don't panic
+		return None
+	else:
+		save_data["error"] = "Save file corrupted"
+		save_data["player_name"] = "BROKEN SAVE!"
+		logger.error("Failed loading save file.")
+		return save_data
 
 def get_index_of_latest_save() -> Optional[int]:
-    times = []
-    for slot_index in range(3):
-        save_path = f"{prepare.SAVE_PATH}{slot_index + 1}.save"
-        save_data = open_save_file(save_path)
-        if save_data is not None:
-            time_of_save = datetime.datetime.strptime(save_data["time"], TIME_FORMAT)
-            times.append((slot_index, time_of_save))
-    if len(times) > 0:
-        s = max(times, key=itemgetter(1))
-        return s[0]
-    else:
-        return None
+	times = []
```

Comment:
PEP8 -- please avoid changing formatting unless it is to bring existing code into compliance